subsys/bluetooth/controller/util/mayfly.c

```diff
@@ -154,10 +159,12 @@ static void dequeue(uint8_t callee_id, uint8_t caller_id, memq_link_t *link,
 		m->_link = link;
 
 		/* reset mayfly state to idle */
+		cpu_dmb();
```

Comment:
I'm not familiar with the `cpu_dmb()`, and it doesn't have any (generic) documentation on what it does or how it works. 

I can see that it marks something as `volatile`, but is that only the next operation, or all operations after `cpu_dmb()` has been present?

Reply:
This is inserting a compiler stronger Data Memory Barrier:
`asm volatile ("" : : : "memory")`

Refer: https://gcc.gnu.org/onlinedocs/gcc/Volatiles.html

FYI, `cpu_dmb()` is Zephyr Controller specific HAL implementation, I see that recently (May'23) Zephyr is providing now a interface `barrier_dsync_fence_full()`  replacing __DSB() instruction which is more a CPU memory barrier instruction in combination with compiler memory barrier but focusing on multi-cpu senarios (SMP). On single core, ___DSB() is not implemented.